unstructured/ingest/pipeline/source.py

```diff
@@ -12,7 +17,34 @@
 
 @dataclass
 class Reader(SourceNode):
-    def run(self, ingest_doc_dict: dict) -> t.Optional[str]:
+    def get_single(self, doc: BaseSingleIngestDoc, ingest_doc_dict: dict) -> str:
+        if (
+            not self.read_config.re_download
+            and doc.filename.is_file()
+            and doc.filename.stat().st_size
+        ):
+            logger.info(f"File exists: {doc.filename}, skipping download")
```

Comment:
I guess with batch there is no real way to "skip download" since at that time we won't know whats in the batch compared to what already exists.

How is batch going to work with incremental ingestion? I guess that part comes before the actual deciding what to download?

Reply:
We don't support that at the moment so my thoughts are right now is lets optimize for what we do support and figure that out when we cross the bridge concerning incremental workflows. We'll need some sort of state when we introduce incremental workflows so we can play around with the data there. We have no guarantee of last modified dates existing on the ES content anyways so we'd most likely have to reingest the entire thing each time.